src/execution/operator/csv_scanner/util/csv_error.cpp

```diff
@@ -0,0 +1,188 @@
+	error << options.ToString();
+	return CSVError(error.str(), CSVErrorType::MAXIMUM_LINE_SIZE);
+}
+
+CSVError CSVError::SniffingError(string &file_path) {
+	std::ostringstream error;
+	// Which column
+	error << "Error when sniffing file \"" << file_path << "\"." << std::endl;
+	error << "CSV options could not be auto-detected. Consider setting parser options manually." << std::endl;
+	return CSVError(error.str(), CSVErrorType::SNIFFING);
+}
+
+CSVError CSVError::UnterminatedQuotesError(const CSVReaderOptions &options, string_t *vector_ptr,
+                                           idx_t vector_line_start, idx_t current_column) {
+	std::ostringstream error;
+	// What is the problematic CSV Line
+	error << "Value with unterminated quote found." << std::endl;
+	error << "Problematic CSV Line (Up to unquoted value):" << std::endl;
+	for (; vector_line_start < current_column; vector_line_start++) {
+		error << vector_ptr[vector_line_start].GetString();
+		if (vector_line_start < current_column - 1) {
+			// we are not in the last line, add the delimiter
+			error << options.dialect_options.state_machine_options.delimiter.GetValue();
+		}
+	}
+	error << std::endl;
+	// What were the options
+	error << options.ToString();
+	return CSVError(error.str(), CSVErrorType::UNTERMINATED_QUOTES);
+}
+
+CSVError CSVError::IncorrectColumnAmountError(const CSVReaderOptions &options, string_t *vector_ptr,
+                                              idx_t vector_line_start, idx_t actual_columns) {
+	std::ostringstream error;
+	// How many columns were expected and how many were found
+	error << "Expected Number of Columns: " << options.dialect_options.num_cols << " Found: " << actual_columns
+	      << std::endl;
+	// What is the problematic CSV Line
+	error << "Problematic CSV Line:" << std::endl;
+	for (; vector_line_start < actual_columns; vector_line_start++) {
+		//		error << vector_ptr[vector_line_start].GetString();
+		if (vector_line_start < actual_columns - 1) {
+			// we are not in the last line, add the delimiter
+			error << options.dialect_options.state_machine_options.delimiter.GetValue();
+		}
+	}
+	error << std::endl;
+	// What were the options
+	error << options.ToString();
+	return CSVError(error.str(), CSVErrorType::INCORRECT_COLUMN_AMOUNT);
+}
+
+bool CSVErrorHandler::PrintLineNumber(CSVError &error) {
+	switch (error.type) {
+	case CSVErrorType::CAST_ERROR:
+	case CSVErrorType::UNTERMINATED_QUOTES:
+	case CSVErrorType::INCORRECT_COLUMN_AMOUNT:
+	case CSVErrorType::MAXIMUM_LINE_SIZE:
+		return true;
+	default:
+		return false;
+	}
+}
+
+idx_t CSVErrorHandler::GetLine(LinesPerBoundary &error_info) {
+	idx_t current_line = 1 + error_info.lines_in_batch; // We start from one, since the lines are 1-indexed
+	for (idx_t boundary_idx = 0; boundary_idx < error_info.boundary_idx; boundary_idx++) {
+		bool batch_done = false;
+		while (!batch_done) {
+			unique_ptr<lock_guard<mutex>> parallel_lock;
```

Comment:
I just realized this is unused, we initialize the unique_ptr as nullptr but don't do anything further with it